tea.py

```diff
@@ -0,0 +1,460 @@
+            print(
+                "Warning: Missing 'Negative current collector density [kg.m-3]', 8960 [kg.m-3] has been used."
+            )
+        else:
+            self.results["Negative current collector density [kg.m-3]"] = self.parameter_values.get("Negative current collector density [kg.m-3]") or self.results.get("Negative current collector density [kg.m-3]")
+        if self.results.get("Positive current collector density [kg.m-3]") is None and self.parameter_values.get("Positive current collector density [kg.m-3]") is None:
+            self.results["Positive current collector density [kg.m-3]"] = 2700 # [kg.m-3]
+            print(
+                "Warning: Missing 'Positive current collector density [kg.m-3]', 2700 [kg.m-3] has been used."
+            )
+        else:
+            self.results["Positive current collector density [kg.m-3]"] = self.parameter_values.get("Positive current collector density [kg.m-3]") or self.results.get("Positive current collector density [kg.m-3]")
+        if self.results.get("Electrolyte density [kg.m-3]") is None and inputs.get("Electrolyte density [kg.m-3]") is None:
+            self.results["Electrolyte density [kg.m-3]"] = 1276 # [kg.m-3] google: (EC:EMC 1:1 1M LiPF6 2% FEC) better just use 1300?
+            print(
+                "Warning: Missing 'Electrolyte density [kg.m-3]', 1276 [kg.m-3] has been used."
+            )
+        else:
+            self.results["Electrolyte density [kg.m-3]"] = inputs.get("Electrolyte density [kg.m-3]") or self.results.get("Electrolyte density [kg.m-3]")
+        self.results["Negative electrode density [kg.m-3]"] = self.parameter_values.get("Negative electrode density [kg.m-3]") or self.results.get("Negative electrode density [kg.m-3]")
+        self.results["Positive electrode density [kg.m-3]"] = self.parameter_values.get("Positive electrode density [kg.m-3]") or self.results.get("Positive electrode density [kg.m-3]")
+        self.results["Separator density [kg.m-3]"] = self.parameter_values.get("Separator density [kg.m-3]") or self.results.get("Separator density [kg.m-3]")
+        
+        # porosities
+        self.results["Negative electrode porosity"] = self.parameter_values.get("Negative electrode porosity") or self.results.get("Negative electrode porosity")
+        self.results["Positive electrode porosity"] = self.parameter_values.get("Positive electrode porosity") or self.results.get("Positive electrode porosity")
+        self.results["Separator porosity"] = self.parameter_values.get("Separator porosity") or self.results.get("Separator porosity")
+        
+        # dry layer true density better sth else than "true"?
+        self.results["Negative electrode composite true density [kg.m-3]"] = self.results.get("Negative electrode composite true density [kg.m-3]") or self.results.get("Negative electrode density [kg.m-3]") / (1 - self.results.get("Negative electrode porosity"))
```

Comment:
shouldn't the electrolyte density factor into this calculation? `wet_elec_density = dry_elec_dens * (1-eps) + elec_desn*eps`